scripts/nix-profile-daemon.sh.in

```diff
@@ -2,7 +2,33 @@
 if [ -n "${__ETC_PROFILE_NIX_SOURCED:-}" ]; then return; fi
 __ETC_PROFILE_NIX_SOURCED=1
 
-export NIX_PROFILES="@localstatedir@/nix/profiles/default $HOME/.nix-profile"
+NIX_LINK=$HOME/.nix-profile
+if [ -n "$XDG_STATE_HOME" ]; then
+    NIX_LINK_NEW="$XDG_STATE_HOME/nix/profile"
+else
+    NIX_LINK_NEW=$HOME/.local/state/nix/profile
```

Comment:
Should this be `NIX_LINK_NEW=$HOME/.local/state/nix/profiles/profile/` ?

Reply:
What does make you think we want to profile/profiles? That looks like a bug to me.

Reply:
@SuperSandro2000 I'm not familiar with the nix source code, but when I do a single user install on a fresh docker ubuntu, nix adds the following sym link:

`~/.nix-profile -> ~/.local/state/nix/profiles/profile`

Do we not want this value to match that? 

If `~/.nix-profile` doesn't exist and `$XDG_STATE_HOME` is not set, we end up with ` NIX_LINK_NEW=$HOME/.local/state/nix/profile` but this link target doesn't exist on my install.

Reply:
It seems like there is both `~/.local/`**`share`**`/nix/profiles/profile` and `~/.local/`**`state`**`/nix/profile`. The former is the actual profile, the latter is a symlink to it. I think this should be simplified by putting the profiles in `state` and removing the useless indirection.

Reply:
Yes this is what I called out in https://github.com/NixOS/nix/pull/5588#issuecomment-1403829182. I think rather than trying to support 4(!) combinations, we should do all the XDG stuff (this and home directories) together so there are just 2 combinations.